tests/test_base.py

```diff
@@ -102,17 +104,31 @@ def is_compatible(pyenv_version: str) -> bool:
             )
 
     @staticmethod
-    def create_pants_ini(*, parent_folder: str, pants_version: Optional[str]) -> None:
-        config = configparser.ConfigParser()
-        if pants_version is not None:
-            config["GLOBAL"] = {
+    def create_pants_config(
+        *, parent_folder: str, pants_version: Optional[str], use_toml: bool = True
+    ) -> None:
+        global_section = (
+            {
                 "pants_version": pants_version,
-                "plugins": "['pantsbuild.pants.contrib.go==%(pants_version)s']",
+                "plugins": ["pantsbuild.pants.contrib.go==%(pants_version)s"],
             }
+            if pants_version is not None
+            else {"plugins": ["pantsbuild.pants.contrib.go"]}
+        )
+        if use_toml:
+            config = {"GLOBAL": global_section}
+            # TODO: string interpolation does not work for TOML when the value comes from the same
+            #  section. This is fixed in Pants 1.26.0.dev1+.
```

Comment:
We discovered this when migrating Toolchain to use TOML. It was fixed in https://github.com/pantsbuild/pants/pull/9114.